pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -208,8 +208,14 @@ func (p *Provider) Version(clusterSpec *cluster.Spec) string {
 	return clusterSpec.VersionsBundle.Tinkerbell.Version
 }
 
-func (p *Provider) EnvMap(_ *cluster.Spec) (map[string]string, error) {
-	return map[string]string{}, nil
+func (p *Provider) EnvMap(spec *cluster.Spec) (map[string]string, error) {
+	environment := map[string]string{}
+	if p.datacenterConfig == nil {
```

Comment:
should this be 
```suggestion
	if p.datacenterConfig != nil {
```

Reply:
[See my other comment.](https://github.com/aws/eks-anywhere/pull/2269#discussion_r882690241)